src/utils/prisma.error.ts

```diff
@@ -0,0 +1,39 @@
+import { Prisma } from '@prisma/client';
+
+const logPrismaError = (e: unknown) => {
+  if (e instanceof Prisma.PrismaClientValidationError) {
```

Comment:
I think this logPrismaError can actually be a switch case. I think that's maybe better since it is easier to add something if needed in the future.

Reply:
I agree with you about it being faster and easier to add new error codes. Thats why we initially chose the switch statement for the PrismaClientKnownRequestError part of the function. The 5 main categories won't change (validation / knownRequest / unknownRequest / rust / initialization) and we considered a switch statement for that.

It turns out that TypeScript has an issue inferring types inside of a switch case using instanceof a class and despite having established that is is an instance of the class, can not see its properties. This means that I need to cast the type in order to access its properties, and the code becomes a bit difficult to read.

I've done my best to tidy it up and I'll push the change.